app/models/user.rb

```diff
@@ -342,14 +342,29 @@ def destroy
     super
   end
 
-  %w[veteran_status military_information payment].each do |emis_method|
-    define_method(emis_method) do
-      emis_model = instance_variable_get(:"@#{emis_method}")
-      return emis_model if emis_model.present?
-
-      emis_model = "EMISRedis::#{emis_method.camelize}".constantize.for_user(self)
-      instance_variable_set(:"@#{emis_method}", emis_model)
-      emis_model
+  military_information_method = 'military_information'
+  define_method(military_information_method) do
+    model = instance_variable_get(:"@#{military_information_method}")
+    return model if model.present?
+
+    form_profile = FormProfile.new(form_id: nil, user: self)
+    model = form_profile.initialize_military_information
+    instance_variable_set(:"@#{military_information_method}", model)
+    model
+  end
+
+  def veteran_status
+    if Flipper.enabled?(:veteran_status_updated)
+      @veteran_status ||= VAProfile::VeteranStatus::Service.new(self)
+    else
+      define_method(emis_method) do
```

Comment:
Why define method in this else block? This seems like it'd generate weird behavior. You're calling `veteran_status`, and if the flipper is not enabled, it defines a method? I don't think that's intentional behavior (and another reason why I hate backend flippers, means you have to test both code paths. In this case I assume this wasn't necessarily a tested or expected code path)

Reply:
It's the legacy code. Yeah, the style is questionable, just a replica of legacy code while testing the new Flipper enabled code above it. Expected to be removed imminently. Discussing now about removing Flipper altogether.